src/algebra/char_p/basic.lean

```diff
@@ -114,23 +115,43 @@ lemma eq_zero [char_zero R] : ring_char R = 0 := (eq R (char_p.of_char_zero R)).
 
 end ring_char
 
+variable {R}
+
+theorem add_pow_prime_eq_pow_add_pow_add_prime_mul_of_commute [semiring R] (p : ℕ)
+  [fact p.prime] (x y : R) (h : commute x y) : ∃ r : R, (x + y) ^ p = x ^ p + y ^ p + p * r :=
```

Comment:
I don't think this should be using `fact` in the lemma statement

Reply:
Yeah I'm not the biggest fact fan, but I have no idea what the rule of thumb is at this point, we have a lot of mathlib lemmas with fact assumptions that don't need to be facts. I'll change this one, but it would be great if we could come up with some guidelines surrounding this and stick to them.
E.g. Only use fact when its necessary for a later TC assumption to fire, or if needed for the statement itself